src/app/tracing/util/map-utils.ts

```diff
@@ -119,49 +146,54 @@ export function createShapeFileLayer(
     return vectorLayer;
 }
 
-function createVectorLayerStyle(styleConfig: ShapeMapStyleConfig): StyleLike {
+function createVectorLayerStyle(styleConfig: ShapeStyleSettings): StyleLike {
     return new Style({
         stroke: new Stroke({
             color: [
-                styleConfig.lineColor.r,
-                styleConfig.lineColor.g,
-                styleConfig.lineColor.b,
+                styleConfig.geojsonBorderColor.r,
+                styleConfig.geojsonBorderColor.g,
+                styleConfig.geojsonBorderColor.b,
             ],
-            width: styleConfig.lineWidth,
+            width: styleConfig.geojsonBorderWidth,
         }),
     });
 }
 
-function getMapLayer(map: ol.Map): BaseLayer | null {
-    const layers = map
+function getMapLayer(map: ol.Map): Array<BaseLayer> {
+    return map
         .getLayers()
         .getArray()
         .filter((layer) => layer.get(LAYER_ID_KEY) === MAP_LAYER_ID);
-    return layers.length > 0 ? layers[0] : null;
 }
 
 export function updateVectorLayerStyle(
     map: ol.Map,
-    styleConfig: ShapeMapStyleConfig,
+    styleConfig: ShapeFileSettings,
 ): void {
-    const vectorLayer = getMapLayer(map);
-    if (vectorLayer instanceof VectorLayer) {
-        const style = createVectorLayerStyle(styleConfig);
-        vectorLayer.setStyle(style);
-    }
+    const mapLayers = getMapLayer(map);
+    // loop through all layers
+    mapLayers.forEach((layer) => {
+        // apply styles only to instances of VectorLayer
+        if (layer instanceof VectorLayer) {
+            const style = createVectorLayerStyle(styleConfig);
+            layer.setStyle(style);
+        }
+    });
 }
 
-export function updateMapType(
-    map: ol.Map,
-    mapConfig: MapConfigWithOptLayout,
-): void {
+export function updateMapType(map: ol.Map, mapConfig: MapViewConfig): void {
```

Comment:
Why do you remove all layers here?
If the user switches e.g. from `MAP_ONLY` to `TILES_AND_SHAPE` it would be better for user experience if the base layer with the tiles is not removed, otherwise the map has to refetch the tiles. Only remove layers if necessary.

By the way: A layer could even be set to invisible ( `layer.setInvisible(true)`)

Reply:
Why do you remove all layers here?
Because we did that before, too. We can work with setVisible, that will result in rewriting most of the code in here, though

Reply:
resolved